content/docs/ops/runbook/rotating-cloudfoundry-diego.md

```diff
@@ -0,0 +1,202 @@
+```
+
+This script will generate all 25 of the certificates for both Diego and
+Cloud Foundry. Since these deployments need to be deployed back-to-back, the
+certificate generation is consolidated in the `cg-deploy-cf` repository.
+
+#### Mind the Security Configuration for Consul schedule
+
+These properties need to be rotated on the same schedule mentioned in the
+_Security Configuration for Consul_ documentation for Cloud Foundry. The
+`consul` VMs need to be updated for both Cloud Foundry and Diego.
+
+- `consul.ca_cert`
+    - Deploy the new CA certificate after the old one for the first deployment.
+      And remove the old CA certificate for the last deployment.
+- `consul.agent_cert`
+    - Deploy this only after a successful deployment of `consul.ca_cert`.
+- `consul.agent_key`
+    - Deploy this only after a successful deployment of `consul.ca_cert`.
+- `consul.server_cert`
+    - Deploy this only after a successful deployment of `consul.ca_cert`.
+- `consul.server_key`
+    - Deploy this only after a successful deployment of `consul.ca_cert`.
+- `consul.encrypt_keys`
+    - Deploy the new encrypt key before the old one for the first deployment.
+      And remove the old encrypt key for the last deployment.
+
+Make sure that Diego deploys after each rotation step for the certificate
+updates from _Security Configuration for Consul_ in order to prevent downtime.
+
+#### Updating certificates in secrets
+
+Replace all the certificates corresponding to the filename and property matching
+in the secrets file, e.g. `certs_dir/hm9000_server.crt` maps to
+`properties.hm9000.server_cert`. Be sure to include the `-----BEGIN
+CERTIFICATE-----` and `-----END CERTIFICATE----- ` and indent the value under
+the YAML stanza to ensure valid YAML syntax.
+
+#### Remaining secrets
+
+The remaining secrets in this file besides certificates can all be rotated at
+once. Do this ahead of the first deployment so that subsequent deployments only
+need to follow the _Security Configuration for Consul_.
+
+### External secrets (external.yml)
+
+External secrets fall into a few categories which range from _rotatable_ with
+coordination with other cloud.gov deployments on one end and _non-rotatable_
+without data loss.
+
+#### Properties used in other deployments
+
+1. `uaa.clients.*.secrets` properties need to be coordinated across various Bosh
+   deployments and cloud.gov apps or service brokers.
+1. `collectd.riemann_server` property for Riemann server address.
+1. `nessus-agent.*` properties for Nessus.
+1. `login.smtp.*` properties for Mandrill.
+
+#### Metadata and properties for external SAML providers
+
+1. `login.providers.*` metadata along with `login.saml.*` properties for
+   certificates and keys.
+    - The metadata comes from our external SAML providers. In order to rotate
+      these coordinate with them to exchange new metadata. The properties for
+      `login.saml.*` can be generate with the following documentation, although
```

Comment:
login.saml cannot currently be rotated without breaking authenticating with properly-configured IDPs.

We are waiting on https://www.pivotaltracker.com/n/projects/997278/stories/142082611 to complete to allow rotation without downtime.  Once that stories done, rotating this will be a multi-step process similar to the consul certs.